maskrcnn_benchmark/layers/dcn/deform_conv_func.py

```diff
@@ -0,0 +1,259 @@
+import torch
+from torch.autograd import Function
+from torch.nn.modules.utils import _pair
+
+from maskrcnn_benchmark import _C
+
+
+class DeformConvFunction(Function):
+
+    @staticmethod
+    def forward(
+        ctx, 
+        input, 
+        offset, 
+        weight,
+        stride=1, 
+        padding=0, 
+        dilation=1, 
+        groups=1, 
+        deformable_groups=1, 
+        im2col_step=64
+    ):
+        if input is not None and input.dim() != 4:
+            raise ValueError(
+                "Expected 4D tensor as input, got {}D tensor instead.".format(
+                    input.dim()))
+        ctx.stride = _pair(stride)
+        ctx.padding = _pair(padding)
+        ctx.dilation = _pair(dilation)
+        ctx.groups = groups
+        ctx.deformable_groups = deformable_groups
+        ctx.im2col_step = im2col_step
+
+        ctx.save_for_backward(input, offset, weight)
+
+        output = input.new_empty(
+            DeformConvFunction._output_size(input, weight, ctx.padding,
+                                            ctx.dilation, ctx.stride))
+
+        ctx.bufs_ = [input.new_empty(0), input.new_empty(0)]  # columns, ones
+
+        if not input.is_cuda:
+            raise NotImplementedError
+        else:
+            cur_im2col_step = min(ctx.im2col_step, input.shape[0])
+            assert (input.shape[0] %
+                    cur_im2col_step) == 0, 'im2col step must divide batchsize'
+            _C.deform_conv_forward(
+                input, 
+                weight, 
+                offset, 
+                output, 
+                ctx.bufs_[0], 
+                ctx.bufs_[1],
+                weight.size(3), 
+                weight.size(2), 
+                ctx.stride[1], 
+                ctx.stride[0],
+                ctx.padding[1], 
+                ctx.padding[0], 
+                ctx.dilation[1],
+                ctx.dilation[0], 
+                ctx.groups, 
+                ctx.deformable_groups,
+                cur_im2col_step
+            )
+        return output
+
+    @staticmethod
```

Comment:
You should ideally make it `once_differentiable` as well, given that there is no support for double backwards here

Reply:
hi @fmassa 
add `once_differentiable` like this?
```
    @staticmethod
    @once_differentiable
    def backward(ctx, grad_output):
```

Reply:
Yes, exactly, that's how it's done for RoIAlign / RoIPool https://github.com/facebookresearch/maskrcnn-benchmark/blob/b318c3ece6960d2884ddfec8bd912a870c99f6c4/maskrcnn_benchmark/layers/roi_align.py#L25-L27